### PR TITLE
Use commit-and-tag-version for Release Management

### PR DESCRIPTION
**Pull request by Chrylo**
- Pushing a tag like release/v0.1.0 will now trigger a release build. The NPM commands should always be used.
- Merging into main will now always trigger a snapshot build of the new version (after bumping). 

Used https://github.com/absolute-version/commit-and-tag-version for the supporting the release process e.g. tagging, committing, bumping versions.

Used https://marketplace.visualstudio.com/items?itemName=bierner.markdown-preview-github-styles for testing the added release markdown doc.

Tested snapshot and release builds here: https://github.com/boschglobal/kuksa-android-sdk/actions/

After merging this:
1) Every Snapshot will be at least a 0.1.1 version
2) We should deploy an official release of 0.1.0

### CODE REVIEW

**Review by wba2hi — CHANGES_REQUESTED** [review]

File: docs/kuksa-sdk_class-diagram.puml [hunk 1]
```diff
@@ -23,27 +23,49 @@ package kuksa {
...
+        + unsubscribe(vssPath: String, Field, PropertyListener)
+        + unsubscribe<T : VssSpecification>(T, Field, VssSpecificationListener<T>)
+    }
+
     class DataBrokerConnection {
         + disconnectListeners: MultiListener<DisconnectListener>
-        + subscribe(List<Property>, PropertyListener)
-        + fetchProperty(Property): GetResponse
-        + updateProperty(Property, Datapoint): SetResponse
+        + subscriptions: Collection<Property>
```
[the remark this answers is not in the document]
Comment (Chrylo) [reply docs/kuksa-sdk_class-diagram.puml:59]: Removed it again. I was already thinking about it.

File: docs/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+1) Execute `npm run commit-release` to prepare the release. This will automatically bump the version to the next correct semantic version, create a CHANGELOG.md file and make a commit.
+2) Check the CHANGELOG.md for wrong entries which may have been commited in the history - We are human after all! :)
+3) Push the commit, create a PR and wait for the merge. 
+
+### Deploying a release
+
+1) Execute `npm run tag-release` on the main branch after the above merge.
+2) Push the tag via 'git push --follow-tags origin feature-x'. The command should be shown in the terminal after step 1. A push of a release tag with the correct format will automatically trigger an offical release GitHub workflow.
+3) Check if the GitHub package was uploaded succesfully.
```
[the remark this answers is not in the document]
Comment (Chrylo) [reply docs/RELEASE.md:28]: My thinking was writing this guide for eclipse repo maintainer (who can also push tags there). The description should still fit then?

File: buildSrc/src/main/kotlin/org/eclipse/kuksa/util/SemanticVersion.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.util
```
[the remark this answers is not in the document]
Comment (Chrylo) [reply buildSrc/src/main/kotlin/org/eclipse/kuksa/util/SemanticVersion.kt:20]: Just renamed the class :P but yes...

File: .github/workflows/deploy-release.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: deployment
```
Comment (wba2hi) [inline .github/workflows/deploy-release.yaml:1]: For a better differentiation, we should rename the tasks differently - right now both of them appear as "deployment" in the Github Actions Log:

maybe by removing the root level name it just falls back to the yaml name, whic hwould be totally appropriate here.

**Comment by wba2hi** [comment]
LGTM

**Comment by wba2hi** [comment]
@SebastianSchildt @lukasmittag
 
Ticket can be merged

**Review by wba2hi — APPROVED** [review]

**Review by SebastianSchildt — APPROVED** [review]

**Pull request merged**